File: backend/models/Question.js
'use strict';
const { DataTypes } = require('sequelize');

module.exports = (sequelize) => {
  const Question = sequelize.define('Question', {
    id: { type: DataTypes.INTEGER, primaryKey: true, autoIncrement: true },
    exam_id: { type: DataTypes.INTEGER, allowNull: false },
    question_text: { type: DataTypes.TEXT, allowNull: false },
    question_type: {
      type: DataTypes.ENUM('mcq', 'true_false', 'short_answer'),
      allowNull: false
    },
    options: { type: DataTypes.JSON, allowNull: true },
    correct_answer: { type: DataTypes.STRING(500), allowNull: true },
    marks: { type: DataTypes.INTEGER, allowNull: false, defaultValue: 1 }
  }, {
    tableName: 'questions',
    underscored: true
  });

  return Question;
};

File: backend/app/models/question.py
from app.extensions import db


class Question(db.Model):
    __tablename__ = "questions"

    question_id = db.Column(db.Integer, primary_key=True)
    exam_id = db.Column(db.Integer, db.ForeignKey("exams.exam_id", ondelete="CASCADE"), nullable=False)
    question_text = db.Column(db.Text, nullable=False)
    question_type = db.Column(db.String(20), nullable=False)
    option_a = db.Column(db.Text)
    option_b = db.Column(db.Text)
    option_c = db.Column(db.Text)
    option_d = db.Column(db.Text)
    correct_answer = db.Column(db.String(5), nullable=False)
    marks = db.Column(db.Integer, nullable=False, default=1)
    order_num = db.Column(db.Integer, nullable=False, default=0)
